File: src/CHTL_JS/Lexer/TokenType.java
package CHTL_JS.Lexer;

public enum TokenType {
    // Keywords
    LISTEN,
    ANIMATE,
    DELEGATE,
    TARGET,
    ROUTER,

    // Symbols
    LBRACE, RBRACE,         // { }
    COLON,                  // :
    COMMA,                  // ,
    SEMICOLON,              // ;
    ARROW,                  // ->

    // Selector Syntax
    L_CURLY_BRACE,          // {{
    R_CURLY_BRACE,          // }}
    SELECTOR,               // e.g., #box, .class, div[0]

    // Literals
    IDENTIFIER,
    STRING,
    NUMBER,

    EOF
}

File: CHTL_JS/CHTLJSLexer/token_type.py
from enum import Enum, auto

class CHTLJSTokenType(Enum):
    # CHTL JS specific tokens
    LBRACE_LBRACE = auto()  # {{
    RBRACE_RBRACE = auto()  # }}
    ARROW = auto()          # ->

    # Re-using some standard token types
    IDENTIFIER = auto()
    LBRACE = auto()
    RBRACE = auto()
    LBRACKET = auto()
    RBRACKET = auto()
    COLON = auto()
    SEMICOLON = auto()
    COMMA = auto()
    EQUALS = auto()

    # Keywords
    ANIMATE = auto()
    DELEGATE = auto()
    VIR = auto()
    ROUTER = auto()

    # For now, we'll treat JS code as opaque identifiers
    JS_CODE = auto()

    EOF = auto()
